src/page/sinfo.rs

```diff
@@ -32,28 +32,37 @@ impl core::fmt::Display for SecInfo {
         match self.class {
             Class::Secs => write!(f, "S"),
             Class::Tcs => write!(f, "T"),
-            Class::Reg => write!(f, "{}", self.flags),
-            Class::Va => write!(f, "V"),
-            Class::Trim => write!(f, "^"),
+            Class::Regular => write!(f, "{}", self.flags),
+            Class::VersionArray => write!(f, "V"),
+            Class::Trimmed => write!(f, "^"),
+            Class::ShadowStackFirst => write!(f, "F"),
+            Class::ShadowStackRest => write!(f, "R"),
         }
     }
 }
 
-impl SecInfo {
-    /// Creates a `SecInfo` instance for regular pages
-    pub const fn reg(flags: Flags) -> Self {
+impl From<Class> for SecInfo {
+    fn from(class: Class) -> Self {
+        let flags = match class {
+            // A CPU constraint
+            Class::Regular => Flags::READ,
+            _ => Flags::empty(),
+        };
```

Comment:
Move this code to `SecInfo::new()`. `SecInfo::from(class)` should be an alias for `SecInfo::new(class, None)`.